debezium-core/src/main/java/io/debezium/jdbc/MainConnectionProvidingConnectionFactory.java

```diff
@@ -5,7 +5,7 @@
  */
 package io.debezium.jdbc;
 
-public interface MainConnectionFactory<T> extends ConnectionFactory<T> {
+public interface MainConnectionProvidingConnectionFactory<T> extends ConnectionFactory<T> {
```

Comment:
maybe you intended to name it MainConnectionProvidingFactory?

Reply:
@ggabora As strange is it might sound this was intentional. I want to keep the `ConnectionFactory` suffix. It is a it bit awkard but I believe it is the most descriptive I was able to come up with.

Reply:
@jpechane got it, thanks

Reply:
@jpechane Please merge so I can create PR's for Db2 and Vitess. Thanks!

Reply:
@ggabora Done!